BST.rb

```diff
@@ -0,0 +1,320 @@
+load "node.rb"
+
+class Bst
+
+	def initialize()
+		@head = nil
+	end	
+
+	def insert(value)
+		if @head==nil
+			@head = Node.new(value)
+		else
+			t = Node.new(value)
+
+			temp = @head
+			pre = nil
+
+			while(temp)
+				pre = temp
+				if temp.value < value
+					temp = temp.right
+				else
+					temp = temp.left
+				end
+			end
+
+
+			if pre.value >= value
+				pre.left = t
+			else
+				pre.right = t
+			end
+		end
+
+	end
+
+	def findMax()
+		prev = nil
+		temp = @head
+
+		while temp
+			prev = temp
+			temp = temp.right
+		end
+
+		return prev.value
```

Comment:
return is not required

prev.value will be returned if that is the last line
try it

Reply:
Done